### PR TITLE
Added support for pluginlib.

### PR DESCRIPTION
**Pull request by ernestmc**
This pull request adds support for building packages that use pluginlib in the Android platform.

Pluginlib relies on dynamically locating and loading library packages which contain the plugin code. The problem is that Android doesn't let us scan the filesystem searching for these libaries.

Basically the solution is to collect all plugin information at build time and construct a helper module that will load this information at startup so pluginlib can access it as soon as it is called. All the plugins are statically linked together with the user code, so no dynamic loading is necessary.

The low level Poco library and Pluginlib need to be patched for this mechanism to work. Also a python helper module was created to run the build time analysis.

### CODE REVIEW

File: get_library.sh [hunk 1]
```diff
@@ -84,7 +84,8 @@ elif [ $1 == 'pcrecpp' ]; then
     URL=https://github.com/brianb/pcre-7.8.git
     COMP='git'
 elif [ $1 == 'poco' ]; then
-    URL=http://pocoproject.org/releases/poco-1.4.6/poco-1.4.6p2.tar.gz
+    #URL=http://pocoproject.org/releases/poco-1.4.6/poco-1.4.6p2.tar.gz
```
Comment (garyservin) [inline get_library.sh]: Can you remove this commented line?

File: get_library.sh [hunk 2]
```diff
@@ -105,6 +106,9 @@ elif [ $1 == 'uuid' ]; then
 elif [ $1 == 'yaml-cpp' ]; then
     URL=https://github.com/creativa77/yaml-cpp.git
```
Comment (garyservin) [inline get_library.sh]: And can you also point to the new github account?

File: get_library.sh [hunk 3]
```diff
@@ -105,6 +106,9 @@ elif [ $1 == 'uuid' ]; then
...
     COMP='git'
+elif [ $1 == 'rospkg' ]; then
+    URL=https://github.com/ros-infrastructure/rospkg.git
```
Comment (garyservin) [inline get_library.sh]: Should we rely on the master? Wouldn't it be better to use a specific commit so we don't depend on external changes?

File: get_library.sh (hunk 1 above)
Comment (ernestmc) [inline get_library.sh]: Done.

File: get_library.sh (hunk 3 above)
Comment (ernestmc) [inline get_library.sh]: Fixed to latest commit.

File: get_library.sh (hunk 2 above)
Comment (ernestmc) [inline get_library.sh]: Done.

File: do_everything.sh [hunk 4]
```diff
@@ -264,6 +281,28 @@ if [[ $skip -ne 1 ]] ; then
 
 fi
 
+# Before build
```
Comment (garyservin) [inline do_everything.sh]: Can you add a message to the user, so we know what the script is doing?

Something like:
```
echo
echo -e '\e[34mBuilding ....\e[39m'
echo
```

Comment (ernestmc) [inline do_everything.sh]: Added.

**Comment by ernestmc** [comment]
@garyservin I've updated this pull request hopefully fixing the build. Can you test it?
Let me know before merging so I can squash everything.

**Comment by ernestmc** [comment]
The last updates to this pull-request fix the pluginlib support and add the ```pluginlib_sample_app``` to test it.
After building you can find ```pluginlib_sample_app/bin/pluginlib_sample_app-debug.apk```. Install it and run it on an Android device. Check the logcat to see the output:
```
$ adb logcat | grep PLUGIN_TEST
D/PLUGIN_TEST( 6097): Starting app.
D/PLUGIN_TEST( 6097): Triangle area: 43.30
D/PLUGIN_TEST( 6097): Square area: 100.00
```

**Comment by ernestmc** [comment]
Rebased, squashed and ready to merge.

**Pull request merged**